src/components/pets/Pets.js

```diff
@@ -32,12 +36,75 @@ const linkData = [
     type: "rabbit",
   },
 ];
+
+const randomPetURL = "https://api.petfinder.com/v2/animals?limit=1&sort=random";
 // const randomIndex = Math.floor(linkData.length * Math.random());
 // const type = linkData[randomIndex].type;
 // const randomPet = { img: Random, type };
 // linkData.push(randomPet);
 
 export default function Pets() {
+  let randomPetImage = "";
+  const { tokenHeaders } = usePetAuth();
+
+  const { error, data: pet } = useSWR(
+    [tokenHeaders ? randomPetURL : null, tokenHeaders],
+    fetcher,
+    {
+      revalidateOnFocus: false,
+      revalidateIfStale: false,
+      revalidateOnReconnect: false,
+    }
+  );
+
+  // assigns randomPetImage according to if a photo is available
+  if (pet && pet.photos && pet.photos[0]) {
+    if (typeof pet.photos[0] === "string") {
+      randomPetImage = pet.photos[0];
+    } else if (pet.photos[0].large) {
+      randomPetImage = String(pet.photos[0].large);
+    } else if (pet.photos[0].medium) {
+      randomPetImage = pet.photos[0].medium;
+    } else {
+      randomPetImage = Placeholder;
+    }
+  } else {
+    randomPetImage = Placeholder;
+  }
+
+  if (error || (pet && !pet.name)) {
+    if (error) {
+      return <h1>An Error Occurred</h1>;
+    }
+    return <h1>No pet data</h1>;
+  }
+  if (!pet && !error) {
+    return (
+      <div className="pet__container">
+        <h1>Adopt Your Buddy</h1>
+        <Row>
+          {linkData.map((pet) => (
+            <AnimalType
+              img={pet.img}
+              type={pet.type}
+              link={`${pet.type}`}
+              key={pet.type}
+            />
+          ))}
+          <Spinner
+            animation="grow"
+            variant="primary"
+            role="status"
+            data-testid="spinner-load"
+            style={{ padding: "100px", margin: "auto auto" }}
```

Comment:
either use the css or bootstrap 5 utilities  [link](https://getbootstrap.com/docs/5.0/utilities/spacing/) in the `className`